docs/architecture/adr-001-initial-QGB-p2p-network.md

```diff
@@ -0,0 +1,195 @@
+# ADR 001: Initial QGB P2P network
+
+## Context
+
+The QGB V2 will rely on a P2P network to exchange attestations signatures. This comes after deciding that using the state will result in state bloat, and using block data will always have an opportunity cost for validators. Thus, we will rely on a P2P network for exchanging attestations' signatures between orchestrators and relayers.
+
+An attestation signature can be identified by the `(nonce,orchestrator,type)` tuple:
+
+- `nonce`: nonce of the attestation
+- `orchestrator`: orchestrator address
+- `type`: data commitment confirm or valset confirm
+
+The issue with using a P2P network is that there is no way of knowing the CID of a signature beforehand. So, relayers will not be able to get the confirms directly from the network.
+
+## Possible solutions
+
+The standard solution is to use Libp2p kadDHT implementation for discovery and routing, then using a Bitswap exchange to publish attestation signatures and get them using CIDs. However, in our case, the CIDs will be unknown until the moment of signature creation making it impossible to rely on Bitswap alone. So, we will need to use another mechanism to know the CIDs or to get them.
```

Comment:
```suggestion
The standard solution is to use libp2p kadDHT implementation for discovery and routing, then using a Bitswap exchange to publish attestation signatures and get them using CIDs. However, in our case, the CIDs will be unknown until the moment of signature creation making it impossible to rely on Bitswap alone. So, we will need to use another mechanism to know the CIDs or to get them.
```

Reply:
we used a few different formats of libp2p, such as libP2P, and should likely just stick to a single one.